nilearn/_utils/estimator_checks.py

```diff
@@ -466,6 +475,25 @@ def check_masker_clean(estimator):
     assert_raises(AssertionError, assert_array_equal, detrended_signal, signal)
 
 
+def check_masker_transformer(estimator):
+    """Replace sklearn _check_transformer for maskers.
+
+    - fit_transform method should work on non fitted estimator
+    - fit_transform should give same result as fit then transform
+    """
+    if accept_niimg_input(estimator):
+        input_img = _img_4d_rand_eye_medium()
+    else:
+        input_img = _make_surface_img(100)
+
+    signal_1 = estimator.fit_transform(input_img)
+
+    estimator = clone(estimator)
```

Comment:
The [doc for `clone`](https://scikit-learn.org/stable/modules/generated/sklearn.base.clone.html) says that if the `random_state` is specified and it is not `integer`, the clone might return a different result. Shall we consider that?

Reply:
this merits a double check but I think we do set an integer to all our random state when testing